File: font/rename_gifs.py
#! /usr/bin/env python3
"""
This script renames the glyph GIF images in a directory to strip the non-code part from the filename
"""

import os
import re
import sys

if __name__ == "__main__":
    s_dir = sys.argv[1]
    sources = [os.path.join(s_dir, p) for p in os.listdir(s_dir) if p.endswith(".GIF")]
    for s in sources:
        fname = os.path.basename(s)
        new_fname = re.sub(r"\..*$", ".GIF", fname)
        if new_fname != fname:
            os.rename(s, os.path.join(s_dir, new_fname))

File: scripts/font/rename-gifs.ts
#!/usr/bin/env bun
/**
 * Strip the descriptive suffix from generated glyph GIFs, leaving `<strcode>.GIF`.
 * Port of font/rename_gifs.py.
 */
import { readdirSync, renameSync } from "node:fs";
import { join } from "node:path";

export function renameGifs(dir: string): void {
  for (const name of readdirSync(dir).filter((f) => f.endsWith(".GIF"))) {
    const newName = name.replace(/\..*$/, ".GIF");
    if (newName !== name) renameSync(join(dir, name), join(dir, newName));
  }
}

if (import.meta.main) {
  const dir = process.argv[2];
  if (!dir) {
    console.error("usage: bun run scripts/font/rename-gifs.ts <dir>");
    process.exit(1);
  }
  renameGifs(dir);
}
